### ISSUE TITLE
Docker Login Fails with "no such host"

### ISSUE THREAD

**Issue by RehanSaeed** [docker/for-aws#97]
### Expected behavior

`docker login docker-registry.bridgeinternationalacademies.com --username ... --password ...`
Works as expected.

### Actual behavior

> Error response from daemon: Get https://docker-registry.bridgeinternationalacademies.com/v2/: dial tcp: lookup docker-registry.bridgeinternationalacademies.com on 10.0.0.10:53: no such host

### Information

I am able to login on my dev machine. Running `nslookup` resolves the DNS name successfully on the host, so something must be wrong with docker:

```
$ nslookup docker-registry.bridgeinternationalacademies.com
nslookup: can't resolve '(null)': Name does not resolve

Name:      docker-registry.bridgeinternationalacademies.com
Address 1: xx.xxx.xxx.xx ec2-xx-xxx-xxx-xx.eu-west-1.compute.amazonaws.com
Address 2: xx.xxx.xxx.xx ec2-xx-xxx-xxx-xx.eu-west-1.compute.amazonaws.com
Address 3: xx.xxx.xxx.xx ec2-xx-xxx-xxx-xx.eu-west-1.compute.amazonaws.com
```

I've also tried adding a new `nameserver 8.8.8.8` entry in `/etc/resolv.conf` just in case but that had no effect. I'm using a domain controller for DNS instead of AWS's DNS system.

```
# Generated by dhcpcd from eth0.dhcp
# /etc/resolv.conf.head can replace this line
domain bridgeinternationalacademies.com
nameserver 10.0.0.10
nameserver 10.0.1.10
nameserver 10.0.2.10
nameserver 10.0.3.10
# /etc/resolv.conf.tail can replace this line
```

Is there anything else I can try.

**Comment by kencochrane** [comment]
@RehanSaeed are you running the nslookup command from a manager that you deployed on docker for AWS? If so, can you post the results from `docker info` on that host?

**Comment by RehanSaeed** [comment]
Yes, Here you go:

```
Containers: 4
 Running: 4
 Paused: 0
 Stopped: 0
Images: 5
Server Version: 17.06.0-ce
Storage Driver: overlay2
 Backing Filesystem: extfs
 Supports d_type: true
 Native Overlay Diff: true
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins:
 Volume: local
 Network: bridge host ipvlan macvlan null overlay
 Log: awslogs fluentd gcplogs gelf journald json-file logentries splunk syslog
Swarm: active
 NodeID: kgptyr0sjfymhnm5vymjjjlte
 Is Manager: true
 ClusterID: v2qo6j5irbo6eeb41c5kpbvaf
 Managers: 3
 Nodes: 3
 Orchestration:
  Task History Retention Limit: 5
 Raft:
  Snapshot Interval: 10000
  Number of Old Snapshots to Retain: 0
  Heartbeat Tick: 1
  Election Tick: 3
 Dispatcher:
  Heartbeat Period: 5 seconds
 CA Configuration:
  Expiry Duration: 3 months
  Force Rotate: 0
 Root Rotation In Progress: false
 Node Address: 10.2.0.178
 Manager Addresses:
  10.2.0.178:2377
  10.2.1.76:2377
  10.2.2.184:2377
Runtimes: runc
Default Runtime: runc
Init Binary: docker-init
containerd version: cfb82a876ecc11b5ca0977d1733adbe58599088a
runc version: 2d41c047c83e09a6d61d464906feb2a2f3c52aa4
init version: 949e6fa
Security Options:
 seccomp
  Profile: default
Kernel Version: 4.9.36-moby
Operating System: Alpine Linux v3.5
OSType: linux
Architecture: x86_64
CPUs: 1
Total Memory: 1.951GiB
Name: ip-10-2-0-178.bridgeinternationalacademies.com
ID: 6UTS:WQMP:SQA3:CC7L:RGVS:PO42:7FZZ:2NIM:34YG:F42Z:EAIP:O2ZR
Docker Root Dir: /var/lib/docker
Debug Mode (client): false
Debug Mode (server): true
 File Descriptors: 79
 Goroutines: 152
 System Time: 2017-08-31T07:45:41.357910681Z
 EventsListeners: 0
Registry: https://index.docker.io/v1/
Labels:
 os=linux
 region=eu-west-1
 availability_zone=eu-west-1a
 instance_type=t2.small
 node_type=manager
Experimental: true
Insecure Registries:
 127.0.0.0/8
Live Restore Enabled: false
```

**Comment by RehanSaeed** [comment]
Just fired up a new swarm using the AWS DNS service and that is able to login successfully. I also tried removing all `nameservers` from my `resolv.conf` and adding `8.8.8.8` but when I do a `docker login` it still errrors with `10.0.0.10:53: no such host` which makes me think docker is not picking up the changes from `resolv.conf`.

I also tried running the docker in docker container and passed it a custom DNS setting:

> docker run -it --dns=8.8.8.8 -v /var/run/docker.sock:/var/run/docker.sock docker

I then checked the `resolve.conf` file inside the container and it was using only 8.8.8.8. Still, this also failed with:

> Error response from daemon: Get https://docker-registry.bridgeinternationalacademies.com/v2/: dial tcp: lookup docker-registry.bridgeinternationalacademies.com on 10.0.0.10:53: no such host

So once again docker is not picking up the DNS settings.
